tox.ini

```diff
@@ -2,5 +2,6 @@
 envlist = py38,py39,310
 
 [testenv]
+extras = dev
```

Comment:
This circulary requires tox in toxenv. I don't think the way the test depepndencies are defined is sane. pytest is defined in the extra nad also in deps.

What about:

 - \[test\] extra with pytest and hypothesis
 - \[dev\] extra with tox and the above as well (or not at all)
 - adapt the readme about executing pytest directly to pip install `.[test]` first